Extend AI section with further risks (#61)
the-teachingRSE-project/end_of_RSEng: future_challenges_for_RSEng.tex

\subsection{Impact of AI and automation}
\label{sec:ai}
\begin{whatis}{}{impact-ai-automation}
RSEs understand how AI models work and how to tune and extend them.
With these skills they can help researchers reap their benefits,

> while clearly knowing their limits.

Reviewer: I've added a line about AI-RSE specialists.

> while clearly knowing their limits.

Author: I agree that the "tune and extend" part is a but too far for every RSE to conquer. I mentioned the "AI-RSE specialists" already in this box, to make everyone happy.

\end{whatis}
Generative AI tools, such as GitHub Copilot~\cite{Friedmann2021},
are being added to software development tools~\cite{Alenezi2025}.
These AI-powered code assistants promise to increase productivity
by automatically generating code, and writing test cases and documentation.
Software developers use them to reduce the number of keystrokes,
focus on higher-level design rather than on idiosyncrasies,
and recall syntax without having to consult the documentation~\cite{Liang2024}.
Code assistants have the potential to help novice or even non-expert programmers
write code~\cite{Feldman2024} thus enabling non-experts to participate in research
and software creation, and fostering greater collaboration across disciplines.
However, when these tools are used in practice, many users report increased cognitive
load and frustration stemming from the difficulty to prompt the system effectively
and from debugging the generated code~\cite{Simkute2025}.

> Many factors can influence AI tools adoption~\cite{Russo2024}.

Reviewer 2: This should say either 'AI tool adoption" or "the adoption of AI tools"
```suggestion
Many factors can influence the adoption of AI tools~\cite{Russo2024}.
```

The potential of AI to be a democratising technology can only be realised
when the algorithmic divide along with the digital divide is addressed~\cite{Yu2020}.
We also note members of the SE community have called for improving LLM literacy
by adapting SE curricula~\cite{Kirova2024}.

Besides the direct impact in research software and further research output,
we see a number of risks.
The workshop participants suggested that the emergence of AI and automation might reduce critical thinking and imagination.
We address this in \autoref{sec:complexity}.
Another risk is that the widespread use of AI tools could make community resources

> such as \eg\ Stack Overflow and Wikipedia (which are used to train AI models) poorer

Reviewer: I used conditional here to soften that claim. My two cents: there are a lot of hot takes on Stack Overflow right now, some like to blame AI, others like to blame pre-existing issues with the platform governance and/or community behavior. There are preprints on the effects of AI on community resources, and their conclusions are nuanced and multi-faceted.

due to fewer incentives to maintain them, and could compromise LLMs via `model collapse'
when AI-generated content is added to these websites and becomes part of the training data~\cite{Shumailov2024}.
Further risks that were later discussed include an intensified replication crisis, a de-democratization of research, and loss of trust in science.
Due to the non-deterministic nature of generative AI, we expect AI-assisted simulations and and data analysis to become less reproducible.
This is not an entirely new issue, and similar concerns exist in the HPC community,
where reproducing simulation data and data analysis workflows presents tough challenges~\cite{Antunes2024}.
Professional RSEs could safeguard reproducibility by designing, evaluating, and monitoring appropriate workflows.
When applicable, standardized and version-controlled models should be preferred
over self-developed models, and self-developed models should be treated like any
other research output (i.e. publish the trained model, training data and training workflow).
Future research will require significant computing and storage resources, as well as useful data and models,

> therefore it will be more driven by institutions able to provide these.

Reviewer 2: ```suggestion
therefore it will be more likely to be driven by institutions that are able to provide these.
```

Finally, an increasing risk is the loss of trust in science,
in particular due to the automation of the peer-reviewing process~\cite{Naddaf2025}
and the malicious use of hidden prompts~\cite{Gibney2025}.
As researchers increase their reliance on AI assistants,
RSEs will be freed from repeatedly crafting similar individual elements,

> but will be spending more of their time in designing and reviewing software written by other humans and AI tools.

Reviewer 2: ```suggestion
but will be spending more of their time designing and reviewing software written by other humans and AI tools.
```

> To address these risks, the RSE of the future could be not only closely overlooking every part

Reviewer 2: ```suggestion
To address these risks, the RSE of the future could be not only closely overseeing every part
```

I think this means overseeing (keeping an eye on / looking after) as opposed to overlooking (missing)?

> To address these risks, the RSE of the future could be not only closely overlooking every part
> of the research software development cycle (ensuring dissemination, archiving, and reproducibility),
> but also contributing to endangered community resources and infrastructure that
> will keep research accessible to the wider scientific community.

Reviewer 3: ```suggestion
To address these risks, the RSE of the future should not only steer every part
of the research software development cycle (ensuring dissemination, archiving, and reproducibility),
but also contribute to endangered community resources and infrastructure that
will keep research accessible to the wider scientific community.
```

minor grammar and word tweak

The potential of AI to be a democratising technology can only be realised
when the algorithmic divide along with the digital divide is addressed~\cite{Yu2020}.

Besides the direct impact in research software and further research output,
we see a number of risks.
The workshop participants suggested that the emergence of AI and automation might reduce critical thinking and imagination.
We address this in \autoref{sec:complexity}.
Another risk is that the widespread use of AI tools will make community resources

> such as \eg\ Stack Overflow and Wikipedia (which are used to train AI models) poorer, due to fewer incentives to maintain them.

Reviewer: I've rephrased your suggestion and added a reference.
